moto/iam/config.py

```diff
@@ -45,10 +48,54 @@ def list_config_service_resources(
             # Filtered roles are now the subject for the listing
             role_list = filtered_roles
 
-        # Pagination logic, sort by role id
-        sorted_roles = sorted(role_list, key=lambda role: role.id)
-        # sorted_role_ids matches indicies of sorted_roles
-        sorted_role_ids = list(map(lambda role: role.id, sorted_roles))
+        if aggregator:
+            # IAM is a little special; Roles are created in us-east-1 (which AWS calls the "global" region)
+            # However, the resource will return in the aggregator (in duplicate) for each region in the aggregator
+            # Therefore, we'll need to find out the regions where the aggregators are running, and then duplicate the resource there
+
+            # In practice, it looks like AWS will only duplicate these resources if you've "used" any roles in the region, but since
+            # we can't really tell if this has happened in moto, we'll just bind this to the regions in your aggregator
+            from moto.config.models import CONFIG_REGIONS
+
+            aggregated_regions = []
+            aggregator_sources = aggregator.get(
+                "account_aggregation_sources"
+            ) or aggregator.get("organization_aggregation_source")
+            for source in aggregator_sources:
+                source_dict = source.__dict__
+                if source_dict["all_aws_regions"]:
+                    aggregated_regions = CONFIG_REGIONS
+                    break
+                for region in source_dict["aws_regions"]:
+                    aggregated_regions.append(region)
+
+            duplicate_role_list = []
+            for region in list(set(aggregated_regions)):
+                for role in role_list:
+                    duplicate_role_list.append(
+                        {
+                            "_id": "{}{}".format(
+                                role.id, region
+                            ),  # this is only for sorting, isn't returned outside of this functin
+                            "type": "AWS::IAM::Role",
+                            "id": role.id,
+                            "name": role.name,
+                            "region": region,
+                        }
+                    )
+
+            # Pagination logic, sort by role id
+            sorted_roles = sorted(duplicate_role_list, key=lambda role: role["_id"])
```

Comment:
Why do you need sorted roles and sorted role ids? With regards to the token selection below you can get the index of the role object, and then pull out the ID.